backend/remote/backend_common.go

```diff
@@ -414,33 +414,37 @@ func (b *Remote) checkPolicy(stopCtx, cancelCtx context.Context, op *backend.Ope
 		case tfe.PolicyHardFailed:
 			return fmt.Errorf(msgPrefix + " hard failed.")
 		case tfe.PolicySoftFailed:
+
+			runOverrideOuputMsg := fmt.Sprintf(
+				"The run needs to be manually overridden or discarded.\n%s\n",
+				fmt.Sprintf(runHeader, b.hostname, b.organization, op.Workspace, r.ID),
+			)
 			if op.Type == backend.OperationTypePlan || op.UIOut == nil || op.UIIn == nil ||
-				op.AutoApprove || !pc.Actions.IsOverridable || !pc.Permissions.CanOverride {
-				return fmt.Errorf(msgPrefix + " soft failed.")
+				!pc.Actions.IsOverridable || !pc.Permissions.CanOverride {
+				return fmt.Errorf(msgPrefix + " soft failed.\n" + runOverrideOuputMsg)
 			}
-		default:
-			return fmt.Errorf("Unknown or unexpected policy state: %s", pc.Status)
-		}
 
-		opts := &terraform.InputOpts{
-			Id:          "override",
-			Query:       "\nDo you want to override the soft failed policy check?",
-			Description: "Only 'override' will be accepted to override.",
-		}
-
-		err = b.confirm(stopCtx, op, opts, r, "override")
-		if err != nil && err != errRunOverridden {
-			return err
-		}
-
-		if err != errRunOverridden {
-			if _, err = b.client.PolicyChecks.Override(stopCtx, pc.ID); err != nil {
-				return generalError("Failed to override policy check", err)
+			if op.AutoApprove {
+				if _, err = b.client.PolicyChecks.Override(stopCtx, pc.ID); err != nil {
+					return generalError(fmt.Sprintf("Failed to override policy check.\n%s", runOverrideOuputMsg), err)
+				}
+			} else {
+				opts := &terraform.InputOpts{
+					Id:          "override",
+					Query:       "\nDo you want to override the soft failed policy check?",
+					Description: "Only 'override' will be accepted to override.",
+				}
+				err = b.confirm(stopCtx, op, opts, r, "override")
+				if err != nil && err != errRunOverridden {
+					return fmt.Errorf(fmt.Sprintf("Failed to override: %s.\n%s", err.Error(), runOverrideOuputMsg))
```

Comment:
Something I overlooked before trying this branch out:

Apparently when confirmation not given, the run is automatically discarded, here:

https://github.com/hashicorp/terraform/blob/e966c492813a1a3cbb20afa214c623a055847199/backend/remote/backend_common.go#L536-L560

With the way this error is handled here, you get a "Failed to override" error along with the extra message that isn't really applicable:

![image](https://user-images.githubusercontent.com/2430490/108796782-2ba48300-754f-11eb-9661-1c6b3a630031.png)

Here's what a 'I didn't confirm the override' looks like on `master`:

![image](https://user-images.githubusercontent.com/2430490/108797065-e896df80-754f-11eb-94c0-fe223847b4f3.png)

We'll want to match this.

Reply:
Aah, good catch! Okay, will update.